app/controllers/logs_searches_controller.rb

```diff
@@ -0,0 +1,25 @@
+class LogsSearchesController < ApplicationController
+  def create
+    @search = LogsSearch.new(search_params)
+
+    if !@search.first_step && @search.valid?
```

Comment:
Don't really understand what this conditional does. 
What is the `first_step` method doing on `@search` and why do we want the `not` value of this and it's validity to be true? 
What does the `by` method do on `@search`?

Reply:
Searching logs is split into two steps, picking what you want to filter by, and entering a search term. This `create` method covers both steps, so there's some slight awkwardness to cover it.  

The `first_step` method is just a flag on the `LogsSearch` model. It's set to `true` if we're on that first, "choose what you want to filter by" step of searching. I would welcome naming suggestions!

The `#by` method is what we're searching by - IP or username. I think I should name it `#filter` instead.

This conditional is a ruby idiom (I believe), sometimes known as "[short circuit evaluation](https://blog.revathskumar.com/2013/05/short-circuit-evaluation-in-ruby.html)". The `@search.valid` part won't be evaluated if `!@search.first_step` is false, so we only check validity on the second step. 

This is to stop error messages showing on the first step (as there's never a `term` filled in on the model at that point.)

I wasn't super happy with the naming/structure here, I'll happily take suggestions! Going to look at it again as part of searching by location, I imagine.